File: android-app/app/src/main/java/com/pranavakshit/ssh/data/ConnectionProfile.kt
package com.pranavakshit.ssh.data

data class ConnectionProfile(
    val id: String = java.util.UUID.randomUUID().toString(),
    val name: String,
    val host: String,
    val port: Int = 22,
    val username: String,
    val keyFilePath: String,
    val passphrase: String? = null,
    val lastConnected: Long = 0
)

File: SSHEasyApp/Models/ConnectionProfile.cs
namespace SSHEasyApp.Models;

public enum AuthMethod
{
    Password,
    PrivateKey
}

public class ConnectionProfile
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string Name { get; set; } = "";
    public string Host { get; set; } = "";
    public int Port { get; set; } = 22;
    public string Username { get; set; } = "";
    public AuthMethod AuthMethod { get; set; } = AuthMethod.Password;

    /// <summary>
    /// DPAPI-encrypted password, stored as Base64 string.
    /// </summary>
    public string EncryptedPassword { get; set; } = "";

    /// <summary>
    /// Absolute path to the private key file.
    /// </summary>
    public string PrivateKeyPath { get; set; } = "";

    /// <summary>
    /// DPAPI-encrypted passphrase for the private key, stored as Base64 string.
    /// </summary>
    public string EncryptedPassphrase { get; set; } = "";

    public DateTime LastConnected { get; set; }
}
